subsys/net/ip/ipv6_nbr.c

```diff
@@ -53,17 +53,32 @@ LOG_MODULE_DECLARE(net_ipv6, CONFIG_NET_IPV6_LOG_LEVEL);
 #define MIN_IPV6_MTU NET_IPV6_MTU
 #define MAX_IPV6_MTU 0xffff
 
+#if defined(CONFIG_NET_IPV6_NBR_CACHE) || defined(CONFIG_NET_IPV6_ND)
+/* Global stale counter, whenever ipv6 neighbor enters into
+ * stale state, stale counter is incremented by one.
+ * When network stack tries to add new neighbor and if table
+ * is full, oldest (oldest stale counter) neighbor in stale
+ * state will be removed from the table and new entry will be
+ * added.
+ */
+static u32_t stale_counter;
+
+static struct k_sem nbr_lock;
+#endif
+
 #if defined(CONFIG_NET_IPV6_ND)
 static struct k_delayed_work ipv6_nd_reachable_timer;
 static void ipv6_nd_reachable_timeout(struct k_work *work);
+static void restart_nd_reachable_timer(struct net_nbr *nbr, s64_t time);
 #endif
 
 #if defined(CONFIG_NET_IPV6_NBR_CACHE)
 
+/* Protocol constants from RFC 4861 Chapter 10 */
 #define MAX_MULTICAST_SOLICIT 3
 #define MAX_UNICAST_SOLICIT   3
-#define DELAY_FIRST_PROBE_TIME K_SECONDS(5) /* RFC 4861 ch 10 */
```

Comment:
Why these useful references were removed?

Reply:
The comment was moved three lines above.